handlers.rb

```diff
@@ -80,38 +81,86 @@ def create_book
     puts 'Book added successfully'
     sleep 0.75
   end
-end
 
-def create_rental
-  puts 'Select a book from the following list by number'
-  @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    book_id = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+
+      person_id = gets.chomp.to_i
+
+      print 'Date: '
+      date = gets.chomp.to_s
 
-  book_id = gets.chomp.to_i
+      rental = Rental.new(date, @books[book_id], @people[person_id])
+      @rentals << rental
 
-  puts 'Select a person from the following list by number (not id)'
-  @people.each_with_index do |person, index|
-    puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      puts 'Rental created successfully'
+      sleep 0.75
+    end
   end
 
-  person_id = gets.chomp.to_i
+  def list_rentals_by_person_id
+    print 'ID of person: '
+    id = gets.chomp.to_i
 
-  print 'Date: '
-  date = gets.chomp.to_s
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
+      sleep 0.75
+    end
+  end
+
+  def store_data
+    File.write('books.json', JSON.generate(@books))
+    File.write('people.json', JSON.generate(@people))
+    File.write('rentals.json', JSON.generate(@rentals))
+  end
 
-  rental = Rental.new(date, @books[book_id], @people[person_id])
-  @rentals << rental
+  def parse_books
+    file = 'books.json'
+    return [] unless File.exist? file
 
-  puts 'Rental created successfully'
-  sleep 0.75
-end
+    data = JSON.parse(File.read(file), create_additions: true)
+    data.each do |book|
+      @books.push(Book.new(book['title'], book['author']))
+    end
+  end
 
-def list_rentals_by_person_id
-  print 'ID of person: '
-  id = gets.chomp.to_i
+  def parse_people
+    file = 'people.json'
+    return [] unless File.exist? file
+
+    JSON.parse(File.read(file)).map do |people|
+      if people['json_class'] == 'Student'
+        student = Student.new(age: people['age'], name: people['name'], parent_permission: people['permission'],
+                              classroom: @classroom.label)
```

Comment:
- [ ] Nice Job on preserving the data, but we have a slight issue on the re-opening the app it render an error about this particular line The code is parsing a JSON file and creating an array of Student objects.
The code is iterating through the array, adding each object to the @people variable.
The code creates a student object and adds it to the @people array. but the why classroom.label is not in your stored def.

![Capture d’écran 2022-03-08 171047](https://user-images.githubusercontent.com/61562626/157278076-80d2c6c1-5806-4d74-8b04-3069cf401c10.png)

Reply:
thanks for your awesome review